### PR TITLE
fix: blob.reload() does not work as intuitively expected

### PR DESCRIPTION
**Pull request by dmitry-fa**

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 1]
```diff
@@ -525,24 +525,38 @@ public boolean exists(BlobSourceOption... options) {
   }
 
   /**
-   * Fetches current blob's latest information. Returns {@code null} if the blob does not exist.
+   * Fetches the latest blob properties. Returns {@code null} if the blob no longer exists.
    *
-   * <p>Example of getting the blob's latest information, if its generation does not match the
-   * {@link Blob#getGeneration()} value, otherwise a {@link StorageException} is thrown.
+   * <p>{@code options} parameter can contain the preconditions. E.g. user may need to get the blob
+   * properties only if the content has not been updated externally. {@code StorageException} with
```
Comment (athakor) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:531]: has not been updated -> `has been not updated`

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 2]
```diff
@@ -525,24 +525,38 @@ public boolean exists(BlobSourceOption... options) {
...
+   * the code {@code 412} is thrown if preconditions fail.
+   *
+   * <p>Example of retrieving the blob's latest information only if the content is not updated
```
Comment (athakor) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:534]: how about this -> `Example of retrieving the latest  information of blob's only if the content's not updated externally.`

File: google-cloud-storage/src/test/java/com/google/cloud/storage/BlobTest.java [hunk 3]
```diff
@@ -129,7 +129,11 @@
           .setRetentionExpirationTime(RETENTION_EXPIRATION_TIME)
           .build();
   private static final BlobInfo BLOB_INFO =
-      BlobInfo.newBuilder("b", "n").setMetageneration(42L).build();
+      BlobInfo.newBuilder("b", "n", 12345678L).setMetageneration(42L).build();
```
Comment (athakor) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/BlobTest.java:132]: what `"b"` and `"n"` indicates here?

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 2 above)
Comment (athakor) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:534]: how about this ->  `Example of retrieving the latest  information of blob's only if the content's not updated externally.`

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 1 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:531]: Both variants are acceptable, by my one is used much often

File: google-cloud-storage/src/test/java/com/google/cloud/storage/BlobTest.java (hunk 3 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/test/java/com/google/cloud/storage/BlobTest.java:132]: "b" for bucket, "n" for name. Please not, those names were before me. I just added generation value.

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 2 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:534]: You variant will be correct if change "of blob's" to "of the blob". I prefer my variant. Thanks for your suggestions.

**Review by athakor — APPROVED** [review]

**Pull request closed without merging**